core/src/main/java/org/commonjava/maven/ext/core/impl/PreparseGroovyManipulator.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.impl;
+
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+import org.apache.maven.repository.DefaultMirrorSelector;
+import org.commonjava.maven.ext.common.ManipulationException;
+import org.commonjava.maven.ext.common.model.Project;
+import org.commonjava.maven.ext.core.ManipulationSession;
+import org.commonjava.maven.ext.core.groovy.InvocationPoint;
+import org.commonjava.maven.ext.core.groovy.InvocationStage;
+import org.commonjava.maven.ext.core.state.GroovyState;
+import org.commonjava.maven.ext.core.state.State;
+import org.commonjava.maven.ext.io.FileIO;
+import org.commonjava.maven.ext.io.ModelIO;
+import org.commonjava.maven.ext.io.PomIO;
+import org.commonjava.maven.ext.io.resolver.GalleyAPIWrapper;
+import org.commonjava.maven.ext.io.resolver.GalleyInfrastructure;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Manipulator} implementation that can resolve a remote groovy file and execute it on executionRoot. Configuration
+ * is stored in a {@link GroovyState} instance, which is in turn stored in the {@link ManipulationSession}.
+ */
+@Named("groovy-injection-preparse")
+@Singleton
```

Comment:
I don't _think_ this needs those annotations?

Reply:
On the one hand, we manually instantiate the class, but on the other, we are still using injection here.

Reply:
I didn't think you were manually instantiating it any more?

Reply:
I am manually instantiating the class itself initially before it gets passed to `ManipulationManager`. But, there is an `@Inject` on the constructor so that the  constructor parameters get set automatically, so I think using `@Named` on the class itself is a good idea now?

Reply:
Can you point me where? I thought you were relying on injection. (Which means the above should stay)

Reply:
Discussed in meeting ; this class is using injection so we do need these.